lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+## Client side caching
+
+Http protocol provides several mechanisms how to cache resources in browser via http headers. To decide which headers are suitable for your use case check Google developers [guide](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) or other [resources](https://www.google.com/search?q=HTTP+caching). This tutorial provides overview how to use these techniques in hapi.
+
+### Cache-Control
+
+`Cache-Control` header tells browser and any intermediate cache if resource is cacheable and for how long. For example `Cache-Control:max-age=30, must-revalidate, private` means that browser can cache resource for 30 seconds and `private` means that it should not be cached by intermediate caches, only by browser. `must-revalidate` means that once it expires it has to request the resource again.
+
+Lets see how we can set this header in hapi:
+
+```javascript
+server.route({
+   path: '/hapi/{ttl?}',
+   method: 'GET',
+   handler: function (request, reply) {
+
+      var response = reply({be: 'hapi'});
+      if (request.params.ttl) {
+         response.ttl(request.params.ttl);
+      }
+   },
+   config: {
+      cache: {
+         expiresIn: 30 * 1000,
+         privacy: 'private'
+      }
+   }
+});
+```
```

Comment:
I generally name my code examples to I can refer to them easier; IE **Listing 1**.